scripts/scil_validate_bids.py

```diff
@@ -188,106 +141,112 @@ def get_data(nSub, dwi, t1s, fs, associations, default_readout, clean):
-            elif clean:
-                return {}
-
-        # Find b0 for topup, take the first one
-        # Check fMAP
-        totalreadout = default_readout
-        fmaps = [fmap for fmap in fmaps if fmap != '']
-        if not fmaps:
-            if 'TotalReadoutTime' in dwi_metadata:
-                totalreadout = dwi_metadata['TotalReadoutTime']
-        else:
-            if isinstance(fmaps[0], list):
-                fmaps = [x for xs in fmaps for x in xs]
-
-            for nfmap in fmaps:
-                nfmap_metadata = get_metadata(nfmap)
-                if 'PhaseEncodingDirection' in nfmap_metadata:
-                    fmap_PE = nfmap_metadata['PhaseEncodingDirection']
-                    fmap_PE = fmap_PE.replace(fmap_PE[0], conversion[fmap_PE[0]])
-
-                    opposite_PE = PE.index(fmap_PE)
-                    if 'TotalReadoutTime' in dwi_metadata:
-                        if 'TotalReadoutTime' in nfmap_metadata:
-                            dwi_RT = dwi_metadata['TotalReadoutTime']
-                            fmap_RT = nfmap_metadata['TotalReadoutTime']
-                            if dwi_RT != fmap_RT and totalreadout == '':
-                                totalreadout = 'error_readout'
-                                topup_fmap[opposite_PE] = 'error_readout'
-                            elif dwi_RT == fmap_RT:
-                                topup_fmap[opposite_PE] = nfmap.path
-                                totalreadout = dwi_RT
-                    else:
-                        topup_fmap[opposite_PE] = nfmap.path
-                        totalreadout = default_readout
-
-        if sbref[index] != '' and len(sbref[index]) == 1:
-            topup_sbref[index] = sbref[index][0].path
-
-    if len(dwi) == 2:
-        if not any(s == '' for s in topup_sbref):
-            topup = topup_sbref
-        elif not any(s == '' for s in topup_fmap):
-            topup = topup_fmap
+
+    if len(dwis) == 2:
+        dwi_path[1] = dwis[1].path
+        bvec_path[1] = layout.get_bvec(dwis[1].path)
+        bval_path[1] = layout.get_bval(dwis[1].path)
+        if 'direction' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['direction']]
+        elif 'PhaseEncodingDirection' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['PhaseEncodingDirection']]
+
+    curr_dwi = dwis[0]
+    dwi_path[0] = curr_dwi.path
+    bvec_path[0] = layout.get_bvec(curr_dwi.path)
+    bval_path[0] = layout.get_bval(curr_dwi.path)
+
+    if 'TotalReadoutTime' in curr_dwi.entities:
+        totalreadout = curr_dwi.entities['TotalReadoutTime']
+
+    if 'session' in curr_dwi.entities:
+        nSess = curr_dwi.entities['session']
+
+    if 'run' in curr_dwi.entities:
+        nRun = curr_dwi.entities['run']
+
+    IntendedForPath = os.path.sep.join(curr_dwi.relpath.split(os.path.sep)[1:])
+    if 'TotalReadoutTime' in curr_dwi.entities:
+        related_files = layout.get(part="mag",
```

Comment:
When supplying only subjects with reverse phase encoded DWI (no epi nor sbref), I get this error :

```
2022-12-22 13:54:32 dinf-0051-19b root[3870836] WARNING Found 2 subject(s)
2022-12-22 13:54:32 dinf-0051-19b root[3870836] WARNING ----------------------------
2022-12-22 13:54:32 dinf-0051-19b root[3870836] WARNING # Validating subject: 032215
2022-12-22 13:54:32 dinf-0051-19b root[3870836] WARNING # Looking for T1 files
Traceback (most recent call last):
  File "/data/dev/scilpy/venv/bin/scil_validate_bids.py", line 11, in <module>
    load_entry_point('scilpy', 'console_scripts', 'scil_validate_bids.py')()
  File "/data/dev/scilpy/scripts/scil_validate_bids.py", line 444, in main
    args.clean))
  File "/data/dev/scilpy/scripts/scil_validate_bids.py", line 182, in get_data
    TotalReadoutTime=totalreadout) +\
  File "/data/dev/scilpy/venv/lib/python3.7/site-packages/bids/layout/layout.py", line 626, in get
    raise ValueError(msg + "If you're sure you want to impose "
ValueError: 'IntendedFor' is not a recognized entity. If you're sure you want to impose this constraint, set invalid_filters='allow'.

Setting `invalid_filters='allow'` in all layout.get() using `IntendedFor` seems to do the trick